prediction_market_agent/agents/microchain_agent/tools.py

```diff
@@ -0,0 +1,53 @@
+from eth_typing import HexStr, HexAddress, ChecksumAddress
+from prediction_market_agent_tooling.markets.agent_market import (
+    AgentMarket,
+    FilterBy,
+    SortBy,
+)
+from prediction_market_agent_tooling.markets.omen.omen import OmenAgentMarket
+from prediction_market_agent_tooling.markets.omen.omen_contracts import OmenConditionalTokenContract, \
+    OmenFixedProductMarketMakerContract
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import OmenSubgraphHandler
+from web3 import Web3
+from web3.types import Wei
+
+
+def address_to_checksum_address(address: str) -> ChecksumAddress:
+    return ChecksumAddress(HexAddress(HexStr(address)))
+
+def get_omen_binary_markets() -> list[OmenAgentMarket]:
+    # Get the 5 markets that are closing soonest
+    markets: list[AgentMarket] = OmenAgentMarket.get_binary_markets(
+        filter_by=FilterBy.OPEN,
+        sort_by=SortBy.CLOSING_SOONEST,
+        limit=5,
+    )
+
+
+def get_omen_binary_market_from_question(market: str) -> OmenAgentMarket:
+    markets = get_omen_binary_markets()
+    for m in markets:
+        if m.question == market:
+            return m
+    raise ValueError(f"Market '{market}' not found")
+
+
+def find_index_set_for_market_outcome(market: OmenAgentMarket, market_outcome: str):
+    try:
+        market_outcome_match = market.outcomes.index(market_outcome)
+        # Index_sets start at 1
+        return market_outcome_match + 1
+    except ValueError as e:
```

Comment:
Is this still WIP? Don't think there should be a try-except here 😛

Reply:
Removed